core/src/main/java/org/jdbi/v3/core/mapper/EnumByOrdinalMapperFactory2.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jdbi.v3.core.mapper;
+
+import java.lang.reflect.Type;
+import java.util.Optional;
+import org.jdbi.v3.core.EnumByOrdinal;
+import org.jdbi.v3.core.config.ConfigRegistry;
+import org.jdbi.v3.core.generic.GenericTypes;
+
+/**
+ * Produces enum column mappers, which map enums from numeric columns according to their {@link Enum#ordinal()}.
+ */
+@EnumByOrdinal
+class EnumByOrdinalMapperFactory2 implements ColumnMapperFactory {
```

Comment:
How come we duplicate all these mapper factories and deprecate the old ones for removal?  Can we just deprecate them "for hiding" and continue to use retrofitted existing implementations?  This is a very high LOC add for a small feature.

Reply:
I duplicated them to avoid tangling/spaghetti. I deprecated the old ones for hiding, and when the time comes you can just hit `Delete` on the old ones and all will be right. I like to sever connections between new and old and leave as little shared space as possible, or at least extract the common parts to an independent space where it can be used by old and new without them touching each other. This way you can just forget about deprecated things and delete them at a glance, they take up physical space but have no bearing on things.

I've never found LOC to be a very meaningful measurement compared to things like coupling, scope size/code piling, or excessive abstractions.

I think we should skip this pr until later if there's this much food for arguments in it.

Reply:
Would extracting ByName and ByOrdinal to independent package-private classes so they don't need to be duplicated satisfy you more?

Reply:
This class likewise becomes redundant if `EnumMapperFactory` implements `QualifiedColumnMapperFactory`